packages/cfd/src/Containers/__tests__/cfd-password-modal.spec.js

```diff
@@ -0,0 +1,513 @@
+        fireEvent.click(ele_cancel_btn);
+        await waitFor(() => {
+            expect(WS.verifyEmail).toHaveBeenCalled();
+        });
+    });
+
+    it('should display password field for user to enter the password', async () => {
+        const props = {
+            ...mock_props,
+            is_cfd_password_modal_enabled: true,
+            is_cfd_success_dialog_enabled: false,
+            account_status: { status: [], category: 'Real' },
+            getAccountStatus: mockFn,
+            platform: 'mt5',
+            error_type: '',
+        };
+
+        render(
+            <Router history={history}>
+                <CFDPasswordModal {...props} />
+            </Router>
+        );
+
+        expect(await screen.findByText(/Enter your DMT5 password to add a DMT5 account/i)).toBeInTheDocument();
+        expect(await screen.findByRole('button', { name: /Add account/i })).toBeInTheDocument();
+        expect(await screen.findByTestId('mt5_password')).toBeInTheDocument();
+    });
+
+    it('should hold the password value provided by the user', async () => {
+        const user_input = 'zo8lAet#2q01Ih';
+        const props = {
+            ...mock_props,
+            is_cfd_password_modal_enabled: true,
+            is_cfd_success_dialog_enabled: false,
+            account_status: { status: [], category: 'Real' },
+            getAccountStatus: mockFn,
+            platform: 'mt5',
+            error_type: 'PasswordError',
+            account_type: { category: 'real', type: 'financial_stp' },
+        };
+
+        render(
+            <Router history={history}>
+                <CFDPasswordModal {...props} />
+            </Router>
+        );
+        const ele_password_field = await screen.findByTestId('mt5_password');
+        fireEvent.change(ele_password_field, { target: { value: user_input } });
+
+        await waitFor(() => {
+            expect(ele_password_field.value).toEqual(user_input);
+        });
+    });
+
+    it('should display error message when password does not meet requirements', async () => {
+        validPassword.mockReturnValue(false);
+        const user_input = 'demo@deriv.com';
+        const props = {
+            ...mock_props,
+            is_cfd_password_modal_enabled: true,
+            is_cfd_success_dialog_enabled: false,
+            account_status: { status: [], category: 'Real' },
+            getAccountStatus: mockFn,
+            platform: 'mt5',
+            email: user_input,
+        };
+
+        render(
+            <Router history={history}>
+                <CFDPasswordModal {...props} />
+            </Router>
+        );
+        const ele_password_input = await screen.findByTestId('mt5_password');
+        fireEvent.change(ele_password_input, { target: { value: user_input } });
+        fireEvent.focusOut(ele_password_input);
+
+        await waitFor(() => {
+            expect(validPassword).toHaveBeenCalled();
+        });
```

Comment:
can we test if error msg is displayed here?

Reply:
Will add the asert